docs/guides/latest/monitoring/builtin.md

```diff
@@ -0,0 +1,383 @@
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: stash-prometheus-server
+  namespace: monitoring
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prometheus
+  template:
+    metadata:
+      labels:
+        app: prometheus
+    spec:
+      serviceAccountName: stash-prometheus-server
+      containers:
+      - name: prometheus
+        image: prom/prometheus:v2.4.3
+        args:
+        - "--config.file=/etc/prometheus/prometheus.yml"
+        - "--storage.tsdb.path=/prometheus/"
+        ports:
+        - containerPort: 9090
+        volumeMounts:
+        - name: prometheus-config-volume
+          mountPath: /etc/prometheus/
+        - name: prometheus-storage-volume
+          mountPath: /prometheus/
+        - name: stash-apiserver-cert
+          mountPath: /etc/prometheus/secret/stash-apiserver-cert
+      volumes:
+      - name: prometheus-config-volume
+        configMap:
+          defaultMode: 420
+          name: stash-prometheus-server-conf
+      - name: prometheus-storage-volume
+        emptyDir: {}
+      - name: stash-apiserver-cert
+        secret:
+          defaultMode: 420
+          secretName: stash-apiserver-cert
+          items: # avoid mounting private key
+          - key: tls.crt
+            path: tls.crt
+```
+
+Notice that, we have mounted `stash-apiserver-cert` secret as a volume at `/etc/prometheus/secret/stash-apiserver-cert` directory.
+
+Now, let's create the deployment,
+
+```console
+$ kubectl apply -f https://github.com/stashed/docs/raw/0.8.3/docs/examples/guides/latest/monitoring/builtin/prom-deployment.yaml
+deployment.apps/stash-prometheus-server created
+```
+
+### Verify Monitoring Metrics
+
+Prometheus server is running on port `9090`. We are going to use [port forwarding](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/) to access Prometheus dashboard. Run following command on a separate terminal,
+
+```console
+$ kubectl port-forward -n monitoring stash-prometheus-server-9ddbf79b6-8l6hk 9090
+Forwarding from 127.0.0.1:9090 -> 9090
+Forwarding from [::1]:9090 -> 9090
+```
+
+Now, we can access the dashboard at `localhost:9090`. Open [http://localhost:9090](http://localhost:9090) in your browser. You should see `pushgateway` and `api` endpoints of `stash-operator` service as targets.
+
+<figure align="center">
+  <img alt="Stash Monitoring Flow" src="/docs/images/latest/monitoring/prom-builtin-target.png">
+<figcaption align="center">Fig: Prometheus dashboard</figcaption>
+</figure>
+
+### Verify Backup and Restore Metrics
+
+This section will show you how Prometheus monitors `backup` and `restore` in stash. For `backup` and `restore` metrics monitoring, you need to take backup and restore of Deployment's data using Stash. You could have to complete the whole backup and restore process by following the deployment's backup/restore user guide [here](docs/guides/latest/workloads/deployment.md).
```

Comment:
When you perform backup or restore using Stash, it will send respective Prometheus metrics. You  can check if metrics are sent successfully by performing backup and restore as described [here]().

A screenshot that shows Prometheus metrics send by Stash backup and restore process is given below,

<figure>